src/BookNavigator/bookmarks/ia-bookmarks.js

```diff
@@ -229,21 +229,23 @@ class IABookmarks extends LitElement {
     return bookmark;
   }
 
-  fetchBookmarks() {
-    return this.api.getAll().then(res => res.text()).then((res) => {
-      let response;
+  async fetchBookmarks() {
+    try {
+      const res = await this.api.getAll();
+      const resText = await res.text();
+      let parsedResponse;
       try {
-        response = JSON.parse(res);
+        parsedResponse = await JSON.parse(resText);
       } catch (e) {
-        response = { error: e.message };
+        parsedResponse = {error : e.message};
       }
-      return response;
-    }).then((response) => {
+
       const {
         success,
         error = 'Something happened while fetching bookmarks.',
         value: bkmrks = [],
-      } = response;
+      } = await parsedResponse;
```

Comment:
The parsedResponse isn't a Promise, so no need for await here.
```suggestion
      } = parsedResponse;
```